src/App.tsx

```diff
@@ -1,14 +1,59 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+import { FilterType } from './types/FilterTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<string>(FilterType.ALL);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then((res) => setTodos(res))
+      .catch((error) => {
+        throw new Error(error.message);
+      });
+  }, []);
+
+  const closeSelectedTodoInfo = () => {
+    setSelectedTodo(null);
+  };
+
+  const filteredTodos = useMemo(() => {
+    let result = todos;
+
+    switch (filterType) {
+      case FilterType.ACTIVE:
+        result = todos.filter((todo) => {
+          return !todo.completed;
+        });
+        break;
+
+      case FilterType.COMPLETED:
+        result = todos.filter((todo) => {
+          return todo.completed;
+        });
+        break;
+
+      default:
+        result = todos;
+    }
+
+    return [...result].filter(
+      (todo) => todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
```

Comment:
No need to spread same as filter in case when query is empty

```suggestion
   if (query) {
     return result.filter(
      (todo) => todo.title.toLowerCase().includes(query.toLowerCase()),
      );
   }
   
   return result;   
```